src/crystal/system/win32/file_info.cr

```diff
@@ -27,8 +27,8 @@ struct Crystal::System::FileInfo < ::File::Info
     @reparse_tag = LibC::DWORD.new(0)
   end
 
-  def size : UInt64
-    (@file_attributes.nFileSizeHigh.to_u64 << 32) | @file_attributes.nFileSizeLow.to_u64
+  def size : Int64
+    ((@file_attributes.nFileSizeHigh.to_u64 << 32) | @file_attributes.nFileSizeLow.to_u64).to_i64
```

Comment:
instead of converting to u64 shifting and then converting to i64 can we just convert to i64?

```suggestion
    (@file_attributes.nFileSizeHigh.to_i64 << 32) | @file_attributes.nFileSizeLow.to_i64
```

Reply:
Maybe. I feel my code is safer because the conversion is done at the last moment.

Reply:
I mostly thought about it for performance reasons if there a test we could write for the bug you are worried about?